tools/jvm/src/main/scala/org/scalajs/core/tools/optimizer/ScalaJSClosureOptimizer.scala

```diff
@@ -85,28 +85,37 @@ class ScalaJSClosureOptimizer {
     }
 
     logTime(logger, "Closure: Write result") {
-      writeResult(result, compiler, cfg.output)
+      writeResult(result, compiler, cfg)
     }
   }
 
   private def writeResult(result: Result, compiler: ClosureCompiler,
-      output: WritableVirtualJSFile): Unit = {
+      cfg: Config): Unit = {
+    val output = cfg.output
 
     val outputContent = if (result.errors.nonEmpty) ""
-    else "(function(){'use strict';" + compiler.toSource + "}).call(this);\n"
+    else "(function(){'use strict';\n" + compiler.toSource + "\n}).call(this);\n"
 
     val sourceMap = Option(compiler.getSourceMap())
 
+    val (header, footer) = cfg.customOutputWrapper
+
     // Write optimized code
     val w = output.contentWriter
     try {
+      if (header != "")
+        w.write(header + "\n")
```

Comment:
Consider first computing
```scala
val (header0, footer0) = ...
val header = if (header0 == "") "" else header0 + "\n"
val footer = if (footer0 == "") "" else footer0 + "\n"
```
and then this will be just `w.write(header)`, but most importantly the computation of `offsetLine` down below will be simplified. I think that should be more readable.